tests/utils/test_atomic_write.py

```
@@ -0,0 +1,300 @@
+    def test_temp_file_removed_when_replace_fails(
+        self, tmp_path: Path, monkeypatch: pytest.MonkeyPatch
+    ) -> None:
+        """If ``os.replace`` fails (e.g. cross-device EXDEV), the temp file
+        must be cleaned up — otherwise repeated failures accumulate garbage
+        next to the target, which operators then confuse for real state.
+        """
+        target = tmp_path / "s.yaml"
+
+        real_replace = os.replace
+
+        def _broken_replace(src: str, dst: str) -> None:
```

Ruff (0.15.11)

[warning] Unused function argument: `src`

(ARG001)